[Text Server] Prevent composite glyphs which incorporate kashida from being used for justification.

bruvzg opened the pull request:
| Before | After |
|---|---|
| <img width="301" alt="Screenshot 2022-08-05 at 18 01 48" src="https://user-images.githubusercontent.com/7645683/183106244-24b10013-1c60-4ae1-90e1-3b647e5afb1b.png"> | <img width="301" alt="Screenshot 2022-08-05 at 18 03 09" src="https://user-images.githubusercontent.com/7645683/183106259-ff391fd6-7225-4b80-bc37-efe2fce60213.png"> |

_In this case, grapheme cluster `U+0640 U+062F` which is rendered as a single glyph, was incorrectly used for justification, since text server was checking only its first character value._

bruvzg committed:
[Text Server] Prevent composite glyphs which incorporate kashida from being used for justification. Update TextServer tests to clean up on fail.

akien-mga commented:
Apparently this causes a leak, LeakSanitizers from GCC and Clang are complaining.

bruvzg commented:
> Apparently this causes a leak, LeakSanitizers from GCC and Clang are complaining.

It's probably because some of the tests fail (probably issues with the test), and it exists without freeing resources.

bruvzg commented:
Updated `TextServer` tests, and changed it to use `CHECK_FALSE_MESSAGE` instead of `TEST_FAIL_COND`, now it should exit normally (instead of doing it immediately) if some test fail, like this - https://github.com/godotengine/godot/runs/7695283538?check_suite_focus=true

akien-mga commented:
Thanks!
